add releasedOn date to new challenges

QuincyLarson opened the pull request:
#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [ ] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
<!-- What type of change does your code introduce? After creating the PR, tick the checkboxes that apply. -->
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask in the Help Contributors room linked above. We're here to help! -->
- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description
<!-- Describe your changes in detail -->

This adds a "releasedOn" date so they will get a "New" tag in the challenge map (though this feature has yet to be implemented in our reactified platform).

I also moved some of our retired backend projects into our "take-home practice projects" section so campers can still build them if they're looking for extra practice.

Finally, I fixed the "isRequired" flag for a couple of the intermediate algorithm challenges that were moved over from advanced algorithm challenges (and were previously un-required)

Commit:
add releasedOn date to new challenges

Commit:
add releasedOn date to new challenges

ChrisD1986 commented:
Seems that's we should update /difficulty!


On Jan 22, 2017 3:14 AM, "Quincy Larson" <notifications@github.com> wrote:

> Pre-Submission Checklist
>
>    - Your pull request targets the staging branch of freeCodeCamp.
>    - Branch starts with either fix/, feature/, or translate/ (e.g.
>    fix/signin-issue)
>    - You have only one commit (if not, squash
>    <http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231>
>    them into one commit).
>    - All new and existing tests pass the command npm test. Use git commit
>    --amend to amend any fixes.
>
> Type of Change
>
>    - Small bug fix (non-breaking change which fixes an issue)
>    - New feature (non-breaking change which adds new functionality)
>    - Breaking change (fix or feature that would change existing
>    functionality)
>    - Add new translation (feature adding new translations)
>
> Checklist:
>
>    - Tested changes locally.
>    - Closes currently open issue (replace XXXX with an issue no): Closes
>    #XXXX
>
> Description
>
> This adds a "releasedOn" date so they will get a "New" tag in the
> challenge map (though this feature has yet to be implemented in our
> reactified platform).
>
> I also moved some of our retired backend projects into our "take-home
> practice projects" section so campers can still build them if they're
> looking for extra practice.
>
> Finally, I fixed the "isRequired" flag for a couple of the intermediate
> algorithm challenges that were moved over from advanced algorithm
> challenges (and were previously un-required)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/freeCodeCamp/freeCodeCamp/pull/12720
> Commit Summary
>
>    - add releasedOn date to new challenges
>
> File Changes
>
>    - *M* seed/challenges/01-responsive-web-design/applied-
>    accessibility.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-0>
>    (26)
>    - *M* seed/challenges/01-responsive-web-design/applied-visual-
>    design.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-1>
>    (60)
>    - *M* seed/challenges/01-responsive-web-design/basic-css.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-2>
>    (2)
>    - *M* seed/challenges/01-responsive-web-design/basic-html-and-
>    html5.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-3>
>    (4)
>    - *M* seed/challenges/01-responsive-web-design/css-flexbox.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-4>
>    (959)
>    - *M* seed/challenges/01-responsive-web-design/responsive-web-
>    design-projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-5>
>    (14)
>    - *M* seed/challenges/01-responsive-web-design/responsive-web-
>    design.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-6>
>    (8)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/basic-data-structures.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-7>
>    (21)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/basic-javascript.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-8>
>    (4)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/debugging.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-9>
>    (50)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/es6.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-10>
>    (31)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/functional-programming.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-11>
>    (102)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/object-oriented-programming.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-12>
>    (30)
>    - *M* seed/challenges/02-javascript-algorithms-and-data-
>    structures/regular-expressions.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-13>
>    (86)
>    - *M* seed/challenges/03-front-end-libraries/front-end-libraries-
>    projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-14>
>    (2)
>    - *M* seed/challenges/03-front-end-libraries/react-and-redux.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-15>
>    (4)
>    - *M* seed/challenges/03-front-end-libraries/react.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-16>
>    (4)
>    - *M* seed/challenges/03-front-end-libraries/redux.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-17>
>    (4)
>    - *M* seed/challenges/03-front-end-libraries/sass.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-18>
>    (13)
>    - *M* seed/challenges/04-data-visualization/data-
>    visualization-projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-19>
>    (4)
>    - *M* seed/challenges/04-data-visualization/data-
>    visualization-with-d3.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-20>
>    (33)
>    - *M* seed/challenges/04-data-visualization/json-apis-and-ajax.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-21>
>    (17)
>    - *M* seed/challenges/05-apis-and-microservices/api-and-
>    microservice-projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-22>
>    (23)
>    - *M* seed/challenges/05-apis-and-microservices/basic-node-and-
>    express.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-23>
>    (118)
>    - *M* seed/challenges/05-apis-and-microservices/managing-
>    packages-with-npm.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-24>
>    (56)
>    - *M* seed/challenges/05-apis-and-microservices/mongodb-and-
>    mongoose.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-25>
>    (54)
>    - *M* seed/challenges/06-information-security-and-
>    quality-assurance/advanced-express-tools.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-26>
>    (33)
>    - *M* seed/challenges/06-information-security-and-
>    quality-assurance/information-security-with-helmetjs.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-27>
>    (16)
>    - *M* seed/challenges/06-information-security-and-
>    quality-assurance/quality-assurance-and-information-
>    security-projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-28>
>    (10)
>    - *M* seed/challenges/06-information-security-and-
>    quality-assurance/quality-assurance-and-testing-with-chai.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-29>
>    (33)
>    - *M* seed/challenges/08-coding-interview-questions-and-take-
>    home-assignments/coding-interview-data-structure-questions.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-30>
>    (60)
>    - *M* seed/challenges/08-coding-interview-questions-and-take-
>    home-assignments/take-home-interview-projects.json
>    <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720/files#diff-31>
>    (185)
>
> Patch Links:
>
>    - https://github.com/freeCodeCamp/freeCodeCamp/pull/12720.patch
>    - https://github.com/freeCodeCamp/freeCodeCamp/pull/12720.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/freeCodeCamp/freeCodeCamp/pull/12720>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AVHZedr_4SFMUOW8Wpj2bDDwrphmLIvdks5rUx3lgaJpZM4LqU8i>
> .
>

Commit:
fix a few challenge property omission

Commit:
fix a few challenge property omission

systimotic commented:
@QuincyLarson 
1. Merge conflicts 😢 
2. The last commit seems to be doing something completely different from the rest of this PR. I think it would probably be easier to QA if they were separate PRs (that may have even been your intention).

QuincyLarson commented:
@systimotic I think this is ready. Please squash these commits when you merge this.

Commit:
Fixes multiple flex issues

Commit:
add challenges I forgot to add

Commit:
add challenges I forgot to add

no-stack-dub-sack commented on seed/challenges/02-javascript-algorithms-and-data-structures/basic-data-structures.json:
```diff
@@ -721,19 +738,43 @@
         "  // change code below this line  ",
         "",
         "  // change code above this line",
```
@QuincyLarson merging this currently would be problematic I think. See what I've noted below. This would splice 2 challenges together (object tests w/ array challenge), and it removes an additional test that was added to that last object challenge.

The content that you added in here has actually been moved up to the top as the first array challenge (it was just posted out of order on the GitHub thread).

no-stack-dub-sack commented on seed/challenges/02-javascript-algorithms-and-data-structures/basic-data-structures.json:
```diff
@@ -721,19 +738,43 @@
+        "We have defined a variable called yourArray; complete the declaration by defining an array of at least 5 elements in length. Your array should contain at least one string, one number, and one boolean."
+      ],
+      "challengeSeed": [
+        "let yourArray = // change code here;"
```
remove this section ☝️ 

that looks like it should do it!

no-stack-dub-sack commented on seed/challenges/02-javascript-algorithms-and-data-structures/basic-data-structures.json:
```diff
@@ -721,19 +738,43 @@
         "  // change code above this line",
-        "}",
-        "",
-        "console.log(addFriend(user, 'Pete'));"
```
keep these 3 lines (724-726)☝️ 


remove here down to next comment 👇

no-stack-dub-sack reviewed (dismissed):
@QuincyLarson merging this as is would be problematic I think. See what I've noted below. This would splice 2 challenges together (object tests w/ array challenge), and it removes an additional test that was added to that last object challenge.

The content that you added in here has actually been moved up to the top as the first array challenge (it was just posted out of order on the GitHub thread).

envy7 committed:
removed all empty translations present in the challenges in the seed files

Greenheart committed:
fix(challenge): update tail section for "Return a Value from a Function with Return".

BrendanSweeny committed:
Reformat first four ES6 challenge instructions

robbawebba committed:
Reformat ES6 arrow function challenges

Dagolin committed:
tw (zh-tw) language update on getting-started

Commit:
Disable loopProtect for non-JavaScript code

erictleung committed:
Format Breadth-First Search for graphs challenge

- Apply appropriate formatting for key words
- Make algorithm image responsive
- Apply code tags around appropriate words
- Remove unnecessary semicolon in seed code
- Fix test with Infinity value
- Add solution to Breadth-First Search

Commit:
make fixes recommended by @no-stack-dub-sack

Commit:
make fixes recommended by @no-stack-dub-sack

Greenheart committed:
fix(challenge): Improve example for "Applied Visual Design: Use the u Tag to Underline Text".

* Update instruction.
* Update `challengeSeed` for following challenges.
* Use strict equality for tests.

Commit:
minor fixes after rebasing staging

Commit:
minor fixes after rebasing staging

Commit:
add logic puzzles and fix sequence of interview section

Commit:
add logic puzzles and fix sequence of interview section

QuincyLarson commented:
@no-stack-dub-sack OK - this should be good to merge. Please take another look when time permits :)

Greenheart committed:
fix(challenge): Update instructions and format for text-transform challenge.

* Improve instruction.
* Format CSS values as table.
* Add additional test.
* Use strict equality check.

Greenheart committed:
Merge pull request #12995 from leonfeng/fix/loop-protect-html

Disable loopProtect for non-JavaScript code

no-stack-dub-sack commented on seed/challenges/02-javascript-algorithms-and-data-structures/basic-data-structures.json:
```diff
@@ -813,11 +809,11 @@
         "console.log(addFriend(user, 'Pete'));"
       ],
       "tests": [
```
The below tests need to be switched around (sorry, the first review may have been a little unclear), the 3 showing as removed are the latest work, they are properly formatted with code tags and the instructions will not match the tests if the third test is omitted. 

Just for the record, even though EverNote may not be doing so well, I love this tool that you recommended!

![image](https://cloud.githubusercontent.com/assets/18563015/22490876/c40cf2a6-e7ed-11e6-9c04-bd07fcf0e936.png)

Other than that, everything looks good to me, and this should be a simple change. I would fix it myself, but not sure how to add to your PR without write access to your repo.

no-stack-dub-sack commented on seed/challenges/02-javascript-algorithms-and-data-structures/basic-data-structures.json:
```diff
@@ -813,11 +809,11 @@
-        "assert.deepEqual((function() { delete user.data.friends; user.data.friends = ['Sam', 'Kira', 'Tomo']; return addFriend(user, 'Pete') })(), ['Sam', 'Kira', 'Tomo', 'Pete'], 'message: <code>addFriend(user, \"Pete\")</code> should return <code>[\"Sam\", \"Kira\", \"Tomo\", \"Pete\"]</code>');"
+        "assert('name' in user && 'age' in user && 'data' in user, 'The user object has name, age, and data keys');",
+        "assert((function() { let L1 = user.data.friends.length; addFriend(user, 'Sean'); let L2 = user.data.friends.length; return (L2 === L1 + 1); })(), 'The addFriend function accepts a user object and a friend as arguments and adds the friend to the array of friends in the user object.');"
       ],
```
The below tests need to be switched around (sorry, the first review may have been a little unclear), the 3 showing as removed are the latest work, they are properly formatted with code tags and the instructions will not match the tests if the third test is omitted. 

Just for the record, even though EverNote may not be doing so well, I love this tool that you recommended!

![image](https://cloud.githubusercontent.com/assets/18563015/22490876/c40cf2a6-e7ed-11e6-9c04-bd07fcf0e936.png)

Other than that, everything looks good to me, and this should be a simple change. I would fix it myself, but not sure how to add to your PR without write access to your repo.

no-stack-dub-sack replied:
The below tests need to be switched around (sorry, the first review may have been a little unclear), the 3 showing as removed are the latest work, they are properly formatted with code tags and the instructions will not match the tests if the third test is omitted. 

Just for the record, even though EverNote may not be doing so well, I love this tool that you recommended!

![image](https://cloud.githubusercontent.com/assets/18563015/22490876/c40cf2a6-e7ed-11e6-9c04-bd07fcf0e936.png)

Other than that, everything looks good to me, and this should be a simple change. I would fix it myself, but not sure how to add to your PR without write access to your repo.

no-stack-dub-sack reviewed (dismissed):
@QuincyLarson I hate to do it to you 😧 , because I know you've been working on this for a while, but this needs one final change....

dhcodes committed:
update bower.json with resolution for jQuery

Greenheart committed:
Merge pull request #13022 from dhcodes/fix/jquery-resolution-bower

update bower.json with resolution for jQuery

Greenheart committed:
Merge pull request #12956 from BrendanSweeny/fix/es6-formatting

Reformatted instructions for the first four ES6 challenges

Commit:
Added a comma to seperate x and y coordinates pairs

Greenheart committed:
Merge pull request #13028 from cdrainxv/fix/short-fix-description

Added a comma to separate x and y coordinates pairs

Greenheart committed:
Merge pull request #13010 from erictleung/fix/format-breadth-first-search-graphs

Format Breadth-First Search for graphs challenge

Greenheart committed:
Merge pull request #12910 from envy7/fix/empty-translations

Removed all empty translations present in the challenges in the seed files

dhcodes committed:
Merge pull request #13017 from Greenheart/fix/text-transform-instruction-and-format

fix(challenge): Update instructions and format for text-transform challenge

Commit:
Allow empty spaces in text between tags

Greenheart committed:
Merge pull request #13035 from ajain17/fix/em_italicizing_jquery

Allow empty spaces in text between tags

Greenheart committed:
fix(challenge): Improved tests for Algorithm "Missing Letters".

Thanks to @breatheloud for reporting and triaging this!
* Improve tests to cover what the instructions actually ask for.
* Update solution to pass new test suite.

no-stack-dub-sack committed:
chore(seed): freeCodeCampify remaining sections of codebase

delbertlegg committed:
Added edge case test for Pig Latin challenge

Added edge case tests for Pig Latin Challenge

JosephLivengood committed:
Updated challenge text formatting to be uniform

raisedadead committed:
Merge pull request #12887 from ajain17/fix/multiple_flexbox_issues

Fixes multiple flex issues

raisedadead committed:
Merge pull request #12963 from robbawebba/fix/beta-arrow-functions-tests

Reformat ES6 arrow function challenges

Commit:
Added semi-colon to end of statements

Greenheart committed:
Merge pull request #13057 from delbertlegg/fix/pig-latin-edge-cases

Added edge case tests for Pig Latin challenge

Greenheart committed:
Merge pull request #13062 from JosephLivengood/fix/regex-challenge-formatting

Updated challenge text formatting to be uniform

Greenheart committed:
Merge pull request #13063 from cdrainxv/fix/add-semicolons

Added semi-colon to end of statements

Greenheart committed:
fix(challenge): formatting of "Remove Items Using splice()".

* Improve order of Array-methods in a sentence.
* Fix preposition.
* Fix tab for challengeSeed.

Greenheart committed:
fix(challenge): stricter tests for "Add Items Using splice()".

dnahol committed:
added semicolon

vcapretz committed:
adds .vscode to gitignore (#13074)

vcapretz committed:
fixes html term to all capslock

dnahol committed:
fix challenge test

Greenheart committed:
Merge pull request #13073 from dnahol/fix/challenge-syntax

Added semicolon to Regular Expressions: Match Beginning String Patterns

Greenheart committed:
Merge pull request #13079 from vcapretz/fix/html-term-all-caps

Fixes html term to all capslock

phapp88 committed:
clarified challenge instructions

Greenheart committed:
Merge pull request #13088 from phapp88/fix/challenge-instructions

clarified challenge instructions for "Basic Data Structures: Access an Array's Contents Using Bracket Notation"

no-stack-dub-sack committed:
Merge pull request #13081 from dnahol/fix/challenge-test

fix(challenge): Decrease the Opacity of An Element

Greenheart committed:
Merge pull request #13055 from no-stack-dub-sack/fix/chore-seed-freeCodeCampify

chore(seed): freeCodeCampify remaining sections of codebase

Manish-Giri committed:
Improve instruction in regex ignore case while matching challenge

Greenheart committed:
Merge pull request #13094 from Manish-Giri/fix/regex-ignore-case-matching

Improve instruction in regex ignore case while matching challenge

Manish-Giri committed:
Improve instruction in regex match single character challenge

Manish-Giri committed:
Fix code formatting in regex find characters with lazy matching

billy-reilly committed:
fix(challenges): Additional test cases for 'Escape Sequences in Strings'

JosephLivengood committed:
Updated NPM tilde test to work

Greenheart committed:
Merge pull request #13105 from billy-reilly/fix/tests-escape-sequences

fix(challenge): Additional test cases for 'Escape Sequences in Strings'

Greenheart committed:
Merge pull request #13096 from Manish-Giri/fix/regex-match-single-character

Improve instruction in regex match single character challenge

Greenheart committed:
Merge pull request #13098 from Manish-Giri/fix/update-code-formatting-regex-lazy-matching

Fix code formatting in regex find characters with lazy matching

Greenheart committed:
fix(challenge): Cleaner assertion for "Copy an Array with slice()".

Manish-Giri committed:
Improve instructions in regex match repeating characters

Greenheart committed:
Merge pull request #13120 from Manish-Giri/fix/regex-match-repeating-characters

Improve instructions in regex match repeating characters

Greenheart committed:
fix(challenge): Format challengeSeed to encourage best practices.

* By re-formatting the seed a bit, I hope to help campers complete
their object definitions before they instantiate anything from them.

Greenheart committed:
fix(challenge): Accept more correct solutions for "Use dot notation"

Now accepts either one or two `console.log()` expressions.

Greenheart committed:
fix(challenge): Improve seed code for "OOP: Use instanceof"

* Temporarily disable jshint to allow expressions resulting in dead code
* Update constructor declaration for consistency.

Greenheart committed:
fix(challenge): Add semicolons to "OOP: Inherit from a Supertype"

Greenheart committed:
fix(challenge): small typo in "OOP: Use closure protection"

dhcodes committed:
Merge pull request #13127 from Greenheart/fix/oop-closure-typo

fix(challenge): small typo in "OOP: Use closure protection"

dhcodes committed:
Merge pull request #13126 from Greenheart/fix/oop-supertype-semicolons

fix(challenge): Add semicolons to "OOP: Inherit from a Supertype"

Greenheart committed:
fix(challenge): Stricter tests for "OOP: Constructor property"

* Also fixed an misleading sentence in the instruction.

Greenheart committed:
fix(challenge): Stricter tests for "OOP: Understand Own Properties".

Greenheart committed:
fix(challenge): Stricter tests for "OOP: Iterate Over All Properties"

systimotic committed:
Simplify truncate a string challenge

Commit:
Added semicolon to end of first statement in challegeSeed for Regular Expressions: Find Characters with Lazy Matching

Greenheart committed:
fix(challenge): Add solutions for OOP-challenges.

* Also fix tests that break when run outside of the browser environment.

Greenheart committed:
Merge pull request #13135 from ankitdwivedi23/fix/add-missing-semicolon

Added semicolon to end of first statement in the challenge "Regular Expressions: Find Characters with Lazy Matching"

Greenheart committed:
fix(challenge): Stricter tests for "OOP: New Object Prototype".

BrendanSweeny committed:
Reduce complexity of the 'scopes with var and let' challenge

Hpauric committed:
Fixed Dataset Link for Bar Chart Project

Removed "1" from the end of the link.

Greenheart committed:
Merge pull request #13144 from Hpauric/patch-1

Fix/Dataset Link for Bar Chart Project

no-stack-dub-sack committed:
Merge pull request #13068 from Greenheart/fix/array-splice-formatting

fix(challenge): formatting of "Remove Items Using splice()".

no-stack-dub-sack committed:
Merge pull request #13072 from Greenheart/fix/array-splice-stricter-tests

fix(challenge): stricter tests for "Add Items Using splice()".

no-stack-dub-sack committed:
Merge pull request #13125 from Greenheart/fix/oop-instanceof

fix(challenge): Improve seed code for "OOP: Use instanceof"

no-stack-dub-sack committed:
Merge pull request #13139 from Greenheart/fix/oop-add-solutions

fix(challenge): Add solutions for OOP-challenges.

no-stack-dub-sack committed:
Merge pull request #13141 from Greenheart/fix/oop-new-prototype-stricter-tests

fix(challenge): Stricter tests for "OOP: New Object Prototype".

no-stack-dub-sack committed:
Merge pull request #13124 from Greenheart/fix/oop-dot-notation-log

fix(challenge): Accept more correct solutions for "OOP: Use dot notation"

no-stack-dub-sack committed:
Merge pull request #13129 from Greenheart/fix/oop-constructor-property

fix(challenge): Stricter tests for "OOP: Constructor property"

no-stack-dub-sack committed:
Merge pull request #13040 from Greenheart/fix/missing-letters-improved-tests

fix(challenge): Improved tests for Algorithm "Missing Letters".

no-stack-dub-sack committed:
Merge pull request #13123 from Greenheart/fix/oop-prototype-properties

fix(challenge): Format challengeSeed to encourage best practices.

no-stack-dub-sack committed:
Merge pull request #13131 from Greenheart/fix/oop-own-props-stricter-tests

fix(challenge): Stricter tests for "OOP: Understand Own Properties".

no-stack-dub-sack committed:
Merge pull request #13132 from Greenheart/fix/oop-all-props-stricter-tests

fix(challenge): Stricter tests for "OOP: Iterate Over All Properties"

JosephLivengood committed:
Fix to test string in challenge 'Serve an HTML file'

no-stack-dub-sack committed:
Merge pull request #13149 from JosephLivengood/fix/node-server-static

Fix to test string in challenge 'Serve an HTML file'

no-stack-dub-sack committed:
Merge pull request #13108 from JosephLivengood/fix/npm-tilde-challenge

fix(challenge): Updated NPM tilde test to work

systimotic committed:
Merge pull request #12929 from Greenheart/fix/return-a-value

fix(challenge): Update tail section for "Return a Value from a Function with Return"

Greenheart committed:
fix(challenge): Fix broken test for "FP: Use filter to extract data".

Greenheart committed:
Merge pull request #13156 from systimotic/fix/truncate-a-string

Simplify truncate a string challenge

no-stack-dub-sack committed:
fix(challenge): various grammar and formatting, .env node challenge

dhcodes committed:
replace Hyperdev references with Gomix

Greenheart committed:
fix(challenge): Fix fatal typo in "FP: use Array.slice()"

* Fixes a typo that prevented the challenge from being completed at all.

HKuz committed:
Revise intro for Basic CSS section

Greenheart committed:
Merge pull request #13169 from HKuz/fix/CSSIntro

Revise intro for Basic CSS section

Quinn-H committed:
Unneccessary semicolon fixed

erictleung committed:
Merge pull request #13175 from Quinn-H/fix/Unneccessary-semicolon

Unnecessary semicolon fixed  in "Functional Programming"

Commit:
Replaced shift() with unshift() in description

Commit:
Reworded descriptive text to refer to present literal match example

JosephLivengood committed:
View rendering and Passport challenges in Advanced Node/Express created (#13177)

erictleung committed:
Merge pull request #13167 from Greenheart/fix/fp-array-slice-fatal-typo

fix(challenge): Fix fatal typo in "FP: use Array.slice()"

Manish-Giri committed:
Remove hover state requirement

raisedadead committed:
fix: Update Gomix handler

Manish-Giri committed:
Remove code tags from challenge seed

raisedadead committed:
Merge pull request #12982 from Dagolin/translate/zh-TW-translation-getting-started

Translation of Getting Started in Tradational Chinese (zh-tw)

Bouncey committed:
Remove gltter-sidecar due to vunerability

erictleung committed:
Merge pull request #13159 from Greenheart/fix/fp-filter-broken-test

fix(challenge): Fix broken test for "FP: Use filter to extract data".

dhcodes committed:
Merge pull request #13176 from cdrainxv/fix/short-fix-unshift

Replaced shift() with unshift() in description

erictleung committed:
Merge pull request #13178 from cdrainxv/fix/change-description-wording

Reworded descriptive text to refer to present literal match example.

Manish-Giri committed:
Remove requirement of i flag in regex

erictleung committed:
Merge pull request #13113 from Greenheart/fix/array-slice-cleaner-assertion

fix(challenge): Cleaner assertion for "Copy an Array with slice()".

erictleung committed:
Merge pull request #13012 from Greenheart/fix/remove-u-tag-from-links

fix(challenge): Improve example for "Applied Visual Design: Use the u Tag to Underline Text"

erictleung committed:
Merge pull request #13212 from Manish-Giri/fix/remove-regex-i-flag

Remove requirement of i flag in regex

Manish-Giri committed:
Fix typo in articles

HKuz committed:
Add updated challenge style guide

Greenheart committed:
Merge pull request #13227 from Manish-Giri/fix/fix-typo-articles

Fix typo in articles

Greenheart committed:
Merge pull request #13196 from Manish-Giri/fix/remove-hover-state

Remove hover state requirement

Greenheart committed:
Merge pull request #13228 from HKuz/fix/addChallengeGuide

Add updated challenge style guide

Greenheart committed:
Merge pull request #13143 from BrendanSweeny/fix/improve-es6-scopes-var-let-challenge

Reduce complexity of the 'scopes with var and let' challenge

raisedadead committed:
Merge pull request #13166 from dhcodes/fix/change-hyperdev-to-gomix

replace Hyperdev references with Gomix

raisedadead committed:
feat: Add open to contribution badges

BerkeleyTrue committed:
Merge pull request #13197 from raisedadead/fix/update-gomix-handler

fix: Update Gomix handler

Commit:
Merge pull request #13245 from raisedadead/feature/add-prs-welcome-badge

feat: Add open to contribution badges

becausealice2 committed:
Reword amiguous challenge instructions

JosephLivengood committed:
Social Authentication challenges added to Advanced Node/Express section (#13235)

MagiaSN committed:
Fix description of 2nd test in challenge Inventory Update

Greenheart committed:
Merge pull request #13279 from MagiaSN/fix/inventory-update

Fix description of 2nd test in challenge Inventory Update

Commit:
Edited label text in description and challenge code

systimotic committed:
Merge pull request #13200 from Bouncey/fix/removeGitterSidecar

Chore(package): Remove gitter-sidecar due to vulnerability

rrjoson committed:
Fix media query instructions and test

dhcodes committed:
Merge pull request #13290 from rrjoson/fix/media-query-issue

Fix create media query instructions and test

dhcodes committed:
Merge pull request #13287 from cdrainxv/fix/edit-label-text

Edited label text in description and challenge code

Commit:
Added semicolon to statement

no-stack-dub-sack committed:
Merge pull request #13294 from cdrainxv/fix/add-semicolon

fix(challenge): added semicolon to statement, matching characters challenge

erictleung committed:
Merge pull request #13162 from no-stack-dub-sack/fix/grammar-formatting-env-challenge

fix(challenge): various grammar and formatting, .env node challenge

Manish-Giri committed:
Ensure use of two arguments in function call

erictleung committed:
Merge pull request #13255 from becausealice2/fix/instruction-ambiguity

Reword amiguous challenge instructions

erictleung committed:
Merge pull request #13295 from Manish-Giri/fix/arguments-in-function-call

Ensure use of two arguments in function call

erictleung committed:
Merge pull request #13198 from Manish-Giri/fix/remove-code-tags

Remove code tags from challenge seed

Commit:
fix mis-merge

Commit:
fix mis-merge

Commit:
fix merge conflicts

QuincyLarson commented:
@no-stack-dub-sack Ok - I've made the fix you recommended, then rebased staging in and addressed the conflicts. Please double check, then squash and merge.

systimotic commented:
@QuincyLarson This PR shows 154 commits for me, and still has merge conflicts. Is that intended?

QuincyLarson commented:
@systimotic yes - this branch has become quite ugly. I'm going branch of staging and re-apply the changes manually real quick, then put in a fresh PR.
